src/test/java/com/amazon/ion/bytecode/bin11/bytearray/FloatOpcodeHandlerTest.kt

```diff
@@ -0,0 +1,242 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.bytecode.bin11.bytearray
+
+import com.amazon.ion.TextToBinaryUtils.hexStringToByteArray
+import com.amazon.ion.bytecode.GeneratorTestUtil.assertEqualBytecode
+import com.amazon.ion.bytecode.ir.Instructions
+import com.amazon.ion.bytecode.util.BytecodeBuffer
+import com.amazon.ion.bytecode.util.ConstantPool
+import com.amazon.ion.bytecode.util.unsignedToInt
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+import kotlin.String
+
+class FloatOpcodeHandlerTest {
+    // In these tests, we check the float values themselves written to the bytecode buffer as well as the raw bits
+    // on the floats to ensure that NaN semantics are preserved.
+
+    @Test
+    fun `float0 opcode handler emits correct bytecode`() {
+        val inputByteArray: ByteArray = "6A".hexStringToByteArray()
+        val buffer = BytecodeBuffer()
+
+        var position = 0
+        val opcode = inputByteArray[position++].unsignedToInt()
+        position += Float0OpcodeHandler.convertOpcodeToBytecode(
+            opcode,
+            inputByteArray,
+            position,
+            buffer,
+            ConstantPool(0),
+            intArrayOf(),
+            intArrayOf(),
+            arrayOf()
+        )
+
+        val expectedBytecode = intArrayOf(Instructions.I_FLOAT_F32, 0)
+        assertEqualBytecode(expectedBytecode, buffer.toArray())
+        assertEquals(1, position)
+    }
+
+    @ParameterizedTest
+    @CsvSource(
+        "6B 01 00, 0 01100111 00000000000000000000000, 0.000000059604645", // smallest positive subnormal number
```

Comment:
No need to make assertions about the raw float bits in this file either, for the same reasons as above.